internal/common/function/function.go

```diff
@@ -101,6 +101,33 @@ func ValidateBackupSpec(nonAdminBackup *nacv1alpha1.NonAdminBackup, enforcedBack
 		}
 	}
 
+	if nonAdminBackup.Spec.BackupSpec.StorageLocation != constant.EmptyString {
+		nonAdminBsl := &nacv1alpha1.NonAdminBackupStorageLocation{}
+		err := clientInstance.Get(ctx, types.NamespacedName{
+			Name:      nonAdminBackup.Spec.BackupSpec.StorageLocation,
+			Namespace: nonAdminBackup.Namespace,
+		}, nonAdminBsl)
+		if apierrors.IsNotFound(err) {
+			return fmt.Errorf("NonAdminBackupStorageLocation not found in the namespace: %v", err)
+		} else if err != nil {
+			return fmt.Errorf("NonAdminBackup spec.backupSpec.storageLocation is invalid: %v", err)
+		}
+		if nonAdminBsl.Status.VeleroBackupStorageLocation == nil || nonAdminBsl.Status.VeleroBackupStorageLocation.NACUUID == constant.EmptyString {
```

Comment:
here would not be simpler to just check `nonAdminBsl.Status.VeleroBackupStorageLocation.Status.Phase`?

Reply:
we do this later to compare if the phase is available, but I wonder if it may happen that UUID is empty and that's the check for?